tests/test_pandas.py

```diff
@@ -51,17 +54,25 @@ def test_types_to_datetime() -> None:
 
 def test_types_concat_none() -> None:
     """Test concatenation with None values."""
-    s1: pd.Series | None = None
-    s2: pd.Series = pd.Series([7, -5, 10])
-
-    check(assert_type(pd.concat([s1, s2]), pd.Series), pd.Series)
+    series = pd.Series([7, -5, 10])
+    df = pd.DataFrame({"a": [7, -5, 10]})
 
-    df1: pd.DataFrame | None = None
-    df2: pd.DataFrame = pd.DataFrame({"a": [7, -5, 10]})
+    check(assert_type(pd.concat([None, series]), pd.Series), pd.Series)
+    check(assert_type(pd.concat([None, df]), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(pd.concat([None, series, df], axis=1), pd.DataFrame), pd.DataFrame
+    )
 
-    check(assert_type(pd.concat([df1, df2]), pd.DataFrame), pd.DataFrame)
+    check(assert_type(pd.concat({"a": None, "b": series}), pd.Series), pd.Series)
+    check(assert_type(pd.concat({"a": None, "b": df}), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(pd.concat({"a": None, "b": series, "c": df}, axis=1), pd.DataFrame),
+        pd.DataFrame,
+    )
 
-    check(assert_type(pd.concat([s1, df1, df2]), pd.DataFrame), pd.DataFrame)
+    if TYPE_CHECKING_INVALID_USAGE:
```

Comment:
@twoertwein can you explain what this block means and why this is needed?  Trying to understand what was missing from my tests.

Reply:
This is testing that the type checkers see the code as invalid.  Although I don't think the test is constructed correctly....